### PR TITLE
Sequentially Semi-separable matrix application

### PR DESCRIPTION
**Pull request by hsignarg**
ffpack_sss.inl implements the application of a quasi-separable matrix (represented by its SSS generators) to a dense matrix and the decompression from SSS generators to a dense matrix.
Both functions are tested by test-sss.C

### CODE REVIEW

File: fflas-ffpack/ffpack/ffpack_sss.inl [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+                           s, R + (row - block - 2) * s * ldr, ldr, Fi.zero, Temp1, s);
+            }
+                /* First column if not done already */
+            if (row%2)
+                    /* A_{row + 2, 1} <- Temp1 * Q_{1} */
+                fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, Temp1, s,
+                       Q, ldq, Fi.zero, A + (row + 1) * s * lda, lda);
+        }
+    
+            /* Last row if partial */
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack_sss.inl:347]: Maybe merge this part within the main loop, setting the dimension to rs for the last iteration.

File: fflas-ffpack/ffpack/ffpack_sss.inl [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+                           W + (column - block - 2) * s * ldw, ldw, Temp2, s, Fi.zero, Temp1, s);
+            }
+                /* First row if not done already */
+            if (column%2)
+                    /* A_{1, column + 2} <- U_{1} * Temp1 */
+                fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, U,
+                       ldu, Temp1, s, Fi.zero, A + (column + 1) * s, lda);
+        }
+    
+            /* Last column if partial
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack_sss.inl:440]: Same here

File: tests/test-sss.C [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+             * chooseField returns a pointer, F needs to be passed by its value */
+        Field* F= chooseField<Field>(q,b,seed);
+        if (F==nullptr)
+            return true;
+            /* Initiate random number generator */
+        typename Field::RandIter G(*F,seed++);
+
+            /* Generate generators (which can be cropped a bit) */
+        Element_ptr D = fflas_new (*F, n, s);
+        Element_ptr P = fflas_new (*F, n, s);     // Could be n - s
```
Comment (ClementPernet) [inline tests/test-sss.C]: when changing the dimensions to the suggested ones, the testsuite fails

File: fflas-ffpack/ffpack/ffpack_sss.inl (hunk 1 above)
Comment (hsignarg) [reply fflas-ffpack/ffpack/ffpack_sss.inl]: Fixed

File: fflas-ffpack/ffpack/ffpack_sss.inl (hunk 2 above)
Comment (hsignarg) [reply fflas-ffpack/ffpack/ffpack_sss.inl]: Fixed too

File: tests/test-sss.C (hunk 3 above)
Comment (hsignarg) [reply tests/test-sss.C]: Suggestions were wrong. Generators are now minimized to the correct sizes and the tests pass.

**Comment by ClementPernet** [comment]
LGTM

**Pull request merged**